chainer/function_hooks/timer.py

```diff
@@ -132,11 +133,38 @@ def _humanized_time(self, second):
             second *= 1000.0
         return '%.2f%s' % (second, 'ns')
 
-    def print_report(self, file=sys.stdout):
-        """Prints a summary report of time profiling in functions."""
+    def _align(self, second):
+        """Returns align info: (factor, unit)."""
+        factor = 1
+        for unit in ['sec', 'ms', 'us']:
+            if second * factor >= 1:
+                return factor, unit
+            factor *= 1000.0
+        return factor, 'ns'
+
+    def print_report(self, unit='auto', file=sys.stdout):
+        """Prints a summary report of time profiling in functions.
+
+        Args:
+            unit (str): Supplementary units used for computational times.
+                `sec`, `ms`, `us`, `ns`, `auto`(default) and `auto_foreach`
+                are supported. If `auto`, units of times are aligned to the
+                largest, and if `auto_foreach`, units of times are adjusted for
+                each element.
+        """
         entries = [['FunctionName', 'ElapsedTime', 'Occurrence']]
+        if unit == 'auto':
+            max_time = max(
+                record['elapsed_time'] for record in self.summary().values())
+            factor, unit = self._align(max_time)
+        elif unit != 'auto_foreach':
+            factor = self.table[unit]
         for function_name, record in self.summary().items():
-            elapsed_time = self._humanized_time(record['elapsed_time'])
+            second = record['elapsed_time']
+            if unit != 'auto_foreach':
+                elapsed_time = '%3.2f%s' % (second * factor, unit)
```

Comment:
This formatting logic is also duplicated with `_humanized_time`.

Reply:
Changed to use `_choose_unit` and `elapsed_time = '%3.2f%s' % (second * factor, unit)` anytime.